src/main/java/duke/Task.java

```diff
@@ -0,0 +1,50 @@
+package duke;
+
+/**
+ * Implements a Task object.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task object.
+     * @param description Description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Gets the status of completion of the task.
+     * @return A string representing whether the task is completed.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Gets the description of the task.
+     * @return A string representation of the description.
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Marks the task as completed.
+     */
+    public void markAsDone() {
+        isDone = !isDone;
```

Comment:
Perhaps it would be better to simply change `isDone` to `true`, in the case that the code accidentally triggers the `markAsDone()` again, causing the `isDone` to become false again.